include/usb/usb_ch9.h

```diff
@@ -0,0 +1,246 @@
+	uint8_t iManufacturer;
+	uint8_t iProduct;
+	uint8_t iSerialNumber;
+	uint8_t bNumConfigurations;
+} __packed;
+
+/** USB Unicode (UTF16LE) String Descriptor */
+struct usb_string_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint16_t bString;
+} __packed;
+
+/** USB Association Descriptor */
+struct usb_association_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bFirstInterface;
+	uint8_t bInterfaceCount;
+	uint8_t bFunctionClass;
+	uint8_t bFunctionSubClass;
+	uint8_t bFunctionProtocol;
+	uint8_t iFunction;
+} __packed;
+
+/** USB Standard Configuration Descriptor */
+struct usb_cfg_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint16_t wTotalLength;
+	uint8_t bNumInterfaces;
+	uint8_t bConfigurationValue;
+	uint8_t iConfiguration;
+	uint8_t bmAttributes;
+	uint8_t bMaxPower;
+} __packed;
+
+/** USB Standard Interface Descriptor */
+struct usb_if_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bInterfaceNumber;
+	uint8_t bAlternateSetting;
+	uint8_t bNumEndpoints;
+	uint8_t bInterfaceClass;
+	uint8_t bInterfaceSubClass;
+	uint8_t bInterfaceProtocol;
+	uint8_t iInterface;
+} __packed;
+
+/** USB Standard Endpoint Descriptor */
+struct usb_ep_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bEndpointAddress;
+	uint8_t bmAttributes;
+	uint16_t wMaxPacketSize;
+	uint8_t bInterval;
+} __packed;
+
+/** Descriptor Types defined in Specification Table 9-6 */
+#define USB_DESCR_DEVICE		1
+#define USB_DESCR_CONFIGURATION		2
+#define USB_DESCR_STRING		3
+#define USB_DESCR_INTERFACE		4
+#define USB_DESCR_ENDPOINT		5
+#define USB_DESCR_DEVICE_QUALIFIER	6
+#define USB_DESCR_OTHER_SPEED		7
+#define USB_DESCR_INTERFACE_POWER	8
+#define USB_DESCR_OTG			9
+#define USB_DESCR_DEBUG			10
+#define USB_DESCR_INTERFACE_ASSOC	11
+#define USB_DESCR_BOS			15
+#define USB_DESCR_DEVICE_CAPABILITY	16
+
+/** Class-Specific Descriptor Types defined in USB Common Class Specification */
+#define USB_DESCR_CS_DEVICE		0x21
+#define USB_DESCR_CS_CONFIGURATION	0x22
+#define USB_DESCR_CS_STRING		0x23
+#define USB_DESCR_CS_INTERFACE		0x24
+#define USB_DESCR_CS_ENDPOINT		0x25
+
+/** USB Specification Release Numbers (bcdUSB Descriptor field) */
+#define USB_SRN_1_1			0x0110
+#define USB_SRN_2_0			0x0200
+#define USB_SRN_2_1			0x0210
+
+#define USB_DEC_TO_BCD(dec)	((((dec) / 10) << 4) | ((dec) % 10))
+
+/** USB Device release number (bcdDevice Descriptor field) */
+#define USB_BCD_DRN		(USB_DEC_TO_BCD(KERNEL_VERSION_MAJOR) << 8 | \
+				 USB_DEC_TO_BCD(KERNEL_VERSION_MINOR))
+
+/** USB Standard Configuration Descriptor Characteristics */
+#define USB_SCD_RESERVED	BIT(7)
+#define USB_SCD_SELF_POWERED	BIT(6)
+#define USB_SCD_REMOTE_WAKEUP	BIT(5)
+
+#define USB_SCD_ATTRIBUTES	(USB_SCD_RESERVED |			      \
+				 COND_CODE_1(CONFIG_USB_SELF_POWERED,	      \
+					     USB_SCD_SELF_POWERED, (0)) |     \
+				 COND_CODE_1(CONFIG_USB_DEVICE_REMOTE_WAKEUP, \
+					     USB_SCD_REMOTE_WAKEUP, (0)))
```

Comment:
Please do not remove parentheses around 2nd argument of `COND_CODE_1` macro. Its [documented](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/sys/util.h#L410).